src/main/test/java/com/zufar/onlinestore/product/api/ProductApiImplTest.java

```diff
@@ -0,0 +1,74 @@
+package com.zufar.onlinestore.product.api;
+
+import com.zufar.onlinestore.product.dto.ProductInfoDto;
+import com.zufar.onlinestore.product.dto.ProductListWithPaginationInfoDto;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import java.util.UUID;
+
+import static com.zufar.onlinestore.product.util.ProductPaginationDefaults.PAGE;
+import static com.zufar.onlinestore.product.util.ProductPaginationDefaults.SIZE;
+import static com.zufar.onlinestore.product.util.ProductPaginationDefaults.SORT_ATTRIBUTE;
+import static com.zufar.onlinestore.product.util.ProductPaginationDefaults.SORT_DIRECTION;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(SpringExtension.class)
+class ProductApiImplTest {
```

Comment:
There is no test which cover `throw new ProductNotFoundException(productId)` case for `ProductApiImpl.getProduct` method in this test class.

Reply:
we dont need it in this class